src/community/wfs-templating/src/main/java/org/geoserver/wfstemplating/request/JsonTemplateCallback.java

```diff
@@ -0,0 +1,113 @@
+import javax.xml.namespace.QName;
+import org.geoserver.catalog.*;
+import org.geoserver.config.GeoServer;
+import org.geoserver.ows.AbstractDispatcherCallback;
+import org.geoserver.ows.DispatcherCallback;
+import org.geoserver.ows.Request;
+import org.geoserver.ows.Response;
+import org.geoserver.platform.Operation;
+import org.geoserver.wfs.json.GeoJSONGetFeatureResponse;
+import org.geoserver.wfs.request.GetFeatureRequest;
+import org.geoserver.wfs.request.Query;
+import org.geoserver.wfstemplating.builders.TemplateBuilder;
+import org.geoserver.wfstemplating.builders.impl.RootBuilder;
+import org.geoserver.wfstemplating.configuration.TemplateConfiguration;
+import org.geoserver.wfstemplating.configuration.TemplateIdentifier;
+import org.geoserver.wfstemplating.response.GeoJsonTemplateGetFeatureResponse;
+import org.geotools.feature.NameImpl;
+import org.opengis.filter.Filter;
+
+/**
+ * This {@link DispatcherCallback} implementation checks on operation dispatched event if a json-ld
+ * path has been provided to cql_filter and evaluate it against the {@link TemplateBuilder} tree to
+ * get the corresponding {@link Filter}
+ */
+public class JsonTemplateCallback extends AbstractDispatcherCallback {
+
+    private Catalog catalog;
+
+    private GeoServer gs;
+
+    private TemplateConfiguration configuration;
+
+    public JsonTemplateCallback(GeoServer gs, TemplateConfiguration configuration) {
+        this.gs = gs;
+        this.catalog = gs.getCatalog();
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Operation operationDispatched(Request request, Operation operation) {
+        if ("WFS".equalsIgnoreCase(request.getService())
+                && request.getOutputFormat() != null
+                && (request.getOutputFormat().equals(TemplateIdentifier.JSONLD.getOutputFormat())
+                        || request.getOutputFormat()
+                                .equals(TemplateIdentifier.GEOJSON.getOutputFormat()))) {
+            FeatureTypeInfo typeInfo = null;
+            GetFeatureRequest getFeature = null;
+            getFeature = GetFeatureRequest.adapt(operation.getParameters()[0]);
+            List<Query> queries = getFeature.getQueries();
+            if (getFeature != null && queries != null && queries.size() > 0) {
+                for (int i = 0; i < queries.size(); i++) {
+                    Query q = queries.get(i);
+                    QName type = q.getTypeNames().get(0);
+                    typeInfo =
+                            catalog.getFeatureTypeByName(
+                                    new NameImpl(type.getPrefix(), type.getLocalPart()));
+                    if (typeInfo != null) {
+                        try {
+                            RootBuilder root =
+                                    configuration.getTemplate(typeInfo, request.getOutputFormat());
+                            if (root != null) {
+                                JsonPathVisitor visitor =
```

Comment:
When the formatter starts wrapping variable declaration and simple assignments over two lines, it's a sign you're nesting too much, try to split the method in two or more, keep shorter (but significant, and well named) methods instead.

Reply:
fixed